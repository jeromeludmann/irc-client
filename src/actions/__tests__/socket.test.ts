import {
  connectServer,
  setConnectionEstablished,
  lookup
} from "@app/actions/socket";

describe("socket actions", () => {
  describe("should generate ConnectServerAction", () => {
    it("with the existing connection", () => {
      expect(connectServer("server1", "irc.network", 6667)).toMatchSnapshot();
    });

    it("with a new connection", () => {
      expect(
        connectServer("server1", "irc.network", 6667, true)
      ).toMatchSnapshot();
    });
  });

  it("should generate ConnectionEstablishedAction", () => {
    expect(setConnectionEstablished("server1")).toMatchSnapshot();
  });

  it("should generate LookupFailedAction", () => {
    expect(
      lookup(
        "server1",
        new Error("error while looking up"),
        "address",
        "family",
        "irc.network"
      )
    ).toMatchSnapshot();
  });
});
