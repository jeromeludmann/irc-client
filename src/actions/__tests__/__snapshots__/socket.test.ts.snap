// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`socket actions should generate ConnectServerAction with a new connection 1`] = `
Object {
  "payload": Object {
    "host": "irc.network",
    "newConnection": true,
    "port": 6667,
  },
  "route": Object {
    "channelKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/CONNECT_SERVER",
}
`;

exports[`socket actions should generate ConnectServerAction with the existing connection 1`] = `
Object {
  "payload": Object {
    "host": "irc.network",
    "newConnection": false,
    "port": 6667,
  },
  "route": Object {
    "channelKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/CONNECT_SERVER",
}
`;

exports[`socket actions should generate ConnectionEstablishedAction 1`] = `
Object {
  "route": Object {
    "channelKey": "@status",
    "serverKey": "server1",
  },
  "type": "SOCKET/CONNECTION_ESTABLISHED",
}
`;

exports[`socket actions should generate LookupFailedAction 1`] = `
Object {
  "payload": Object {
    "error": [Error: error while looking up],
  },
  "route": Object {
    "channelKey": "@status",
    "serverKey": "server1",
  },
  "serverKey": "server1",
  "type": "SOCKET/LOOKUP_FAILED",
}
`;
